src/modules/royalty.ts

```diff
@@ -0,0 +1,203 @@
+import { ERC20__factory, Royalty, Royalty__factory } from "@3rdweb/contracts";
+import { BigNumber } from "ethers";
+import { ModuleType } from "../common";
+import {
+  Currency,
+  CurrencyValue,
+  getCurrencyMetadata,
+  getCurrencyValue,
+} from "../common/currency";
+import { Module } from "../core/module";
+import { SplitRecipient } from "../types/SplitRecipient";
+
+export interface ISplitsModule {
+  /**
+   * Returns a list of all recipients with their
+   * respective split percentages.
+   *
+   */
+  getAllRecipients(): Promise<SplitRecipient[]>;
+
+  /**
+   * Get the split percentage of a recipient.
+   *
+   * @param address - The address of the recipient.
+   */
+  getRecipientSplitPercentage(address: string): Promise<SplitRecipient>;
+
+  /**
+   * Returns the amount of royalty available for a recipient
+   * to withdraw in the native currency.
+   *
+   * @param address - The address of the recipient to check the balance of.
+   */
+  balanceOf(address: string): Promise<BigNumber>;
+
+  /**
+   * Returns the amount of royalty available for a recipient
+   * to withdraw in the native currency in a specific currency.
+   *
+   * @param walletAddress - The address of the recipient to check the balance of.
+   * @param tokenAddress - The address of the currency to check the balance in.
+   */
+  balanceOfByToken(
```

Comment:
this name is annoying but I understand it is necessary :D

Reply:
maybe `erc20BalanceOf()` instead actually? Just thinking out loud

Reply:
i think token might be more aligned with our sdk because we now call erc20 tokens token